proto/src/lib.rs

```diff
@@ -184,9 +181,8 @@ where
     }
 
     /// Encodes into a Protobuf-encoded `Vec<u8>`.
-    fn encode_vec(&self) -> Result<Vec<u8>, Error> {
-        let mut wire = Vec::with_capacity(self.encoded_len());
-        self.encode(&mut wire).map(|_| wire)
+    fn encode_vec(&self) -> Vec<u8> {
+        T::from(self.clone()).encode_to_vec()
```

Comment:
I wonder if we can do it without mandatory cloning. However, the present implementation needs a clone too because the conversion to the proto type is done in terms of `From<T>`. There are tradeoffs, and the whole thing is subject for another issue/PR.

Reply:
We could take self by value and leave it up to the caller to clone beforehand if they need to keep access to the original value?

Reply:
Did so ([c8d5fc2](https://github.com/informalsystems/tendermint-rs/pull/1324/commits/c8d5fc24aecf3a913fa13ace52602f4aebc5f6c0)), though to be consistent, I went a bit further and removed some other clones. Hope it works!